model_filereading.cpp

```diff
@@ -64,31 +86,27 @@ void Model::readAtomsFromFile(std::string& filepath){
   std::string line;
   std::ifstream inp_file(filepath);
 
-  int line_counter = 0;
+  int line_counter = 0; // Might not be useful anymore
   // iterate through lines
-  while(getline(inp_file,line)){
-  
-    if(line_counter==0){ // first line contains number of atoms
-      n_atom = std::stoi(line);
-    }
-    
-
-    else if(line_counter == 1){ // comment line (ignore)
-    }
-    
-    else { // subsequent lines contain atom positional data
-      std::vector<std::string> substrings = splitLine(line);
-      for(std::string str : substrings){
-      }
-      // create new atom and add to storage vector (assumes file format: Element_Symbol x y z)
-      Atom at = Atom(std::stod(substrings[1]), 
-                     std::stod(substrings[2]), 
-                     std::stod(substrings[3]), 
-                     substrings[0], 
-                     radii[substrings[0]]);
+  while(getline(inp_file,line)){ // ws to pass empty lines and blank space before atoms if any
+    // divide line into "words"
+    std::vector<std::string> substrings = splitLine(line);
+    // create new atom and add to storage vector if line format corresponds to Element_Symbol x y z
+    if (substrings.size() == 4
+        && isalpha(substrings[0][0])
+        && (isdigit(substrings[1][0]) || ispunct(substrings[1][0])) // ispunct is necessary in case of negative value starting with '-'
+        && (isdigit(substrings[2][0]) || ispunct(substrings[2][0]))
+        && (isdigit(substrings[3][0]) || ispunct(substrings[3][0]))) {
```

Comment:
The obvious issue here is that you only check the first character of each substring. That could lead to issue if the line looks something like:
`Fe 1this 2is 3invalid`
Instead I'll change this to a procedure that checks the whole string.

Apart from that, writing a conditional as a sequence of that many &&s is hard to maintain. In this case checking the first substring isn't really necessary, and you do the same operation for the final three substrings. A for loop is much more appropriate here.